storage/local/persistence.go

```diff
@@ -683,6 +691,11 @@ func (p *persistence) checkpointSeriesMapAndHeads(fingerprintToSeries *seriesMap
 
 	var realNumberOfSeries uint64
 	for m := range iter {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
```

Comment:
The error is never shown in raw fashion.

This is how things look for the user on shutdown:
```
INFO[1172] See you next time!                            source=main.go:240
INFO[1172] Stopping target manager...                    source=targetmanager.go:75
INFO[1172] Stopping rule manager...                      source=manager.go:388
INFO[1172] Rule manager stopped.                         source=manager.go:394
INFO[1172] Stopping notification handler...              source=notifier.go:409
INFO[1172] Stopping local storage...                     source=storage.go:450
INFO[1172] Stopping maintenance loop...                  source=storage.go:452
INFO[1174] Checkpoint canceled.                          source=storage.go:1459
INFO[1174] Maintenance loop stopped.                     source=storage.go:1433
INFO[1174] Stopping series quarantining...               source=storage.go:456
INFO[1174] Series quarantining stopped.                  source=storage.go:1882
INFO[1174] Stopping chunk eviction...                    source=storage.go:460
INFO[1174] Chunk eviction stopped.                       source=storage.go:1128
INFO[1174] Checkpointing in-memory metrics and chunks...  source=persistence.go:619
INFO[1362] Done checkpointing in-memory metrics and chunks in 3m7.83085631s.  source=persistence.go:651
INFO[1362] Checkpointing fingerprint mappings...         source=persistence.go:1512
INFO[1362] Done checkpointing fingerprint mappings in 3.832885ms.  source=persistence.go:1535
INFO[1362] Local storage stopped.                        source=storage.go:477
```